cgfs.py

```diff
@@ -332,21 +463,27 @@ def getattr(self, path, fh=None):
         parts = self.split_path(path)
         file = self.get_file(parts)
 
-        if file.stat is None and len(parts) > 3:
-            submission = self.get_submission(path)
+        if isinstance(file, TempFile):
+            return file.getattr()
 
+        if file.stat is None and len(parts) > 3:
             try:
-                query_path = submission.tld + '/' + '/'.join(parts[3:])
+                submission = self.get_submission(path)
             except CGAPIException as e:
                 handle_cgapi_exception(e)
 
+            query_path = submission.tld + '/' + '/'.join(parts[3:])
```

Comment:
This is used in a few places if I remember correctly, so maybe we should put this in a function? Or maybe we should subclass `Directory` as `CourseDir`, `AssignmentDir` and `SubmissionDir` - then we can use `isinstance` instead of the `DirTypes` enum -  and make this a method on the `SubmissionDir` class?